src/Type.cc

```diff
@@ -1335,7 +1338,11 @@ void RecordType::Create(std::vector<std::optional<ZVal>>& r) const
 
 	for ( int i = 0; i < n; ++i )
 		{
-		auto& init = field_inits[i];
+		auto* init = field_inits[i];
+
+		if ( ! init )
```

Comment:
That `if` is never true. With or without this change there are never `nullptr`s in the vector. With this hunk dropped, the test continues fine.

Reply:
I can fix this up tomorrow and merge if you give a thumbs-up, or otherwise you can push.

Reply:
> That `if` is never true. With or without this change there are never `nullptr`s in the vector. With this hunk dropped, the test continues fine.

That's weird, because I added the test because the debugger showed it was crashing with a null `init`.  Must be that that was a side-effect of the inits-getting-out-of-sync, now fixed elsewhere.

> I can fix this up tomorrow and merge if you give a thumbs-up, or otherwise you can push.

Sure, please go ahead - thanks!